rita_exit/src/database/database_tools.rs

```diff
@@ -636,6 +636,50 @@ fn generate_index_from_subnet(exit_sub: IpNetwork, sub: IpNetwork) -> Result<u64
         .expect("Unalbe to parse biguint into u64"))
 }
 
+/// This function run on startup initializes databases and other missing fields from the previous database schema
+/// for ipv6 support. Existing clients in the previous schema will not have an ipv6 addr assigned, so every client is
+/// given one on startup
+pub fn initialize_exisitng_clients_ipv6() -> Result<(), RitaExitError> {
+    use self::schema::clients::dsl::{clients, internet_ipv6, mesh_ip};
+    let conn = get_database_connection()?;
+
+    // initialize the assigned_ips database
+    let rita_exit = settings::get_rita_exit();
+    let exit_settings = rita_exit.exit_network;
+    let subnet = exit_settings.subnet;
+
+    // If subnet isnt already present in database, create it
+    let subnet_entry = initialize_subnet_datastore(subnet, &conn)?;
+    info!("Subnet Database entry: {:?}", subnet_entry);
+
+    // update all client entries to have an ipv6 addr
+    // 1.) get list of mesh ips
+    // 2.) for each ip, select ipv6 field, if empty set an ipv6, else continue
+    let filtered_list = clients.select(mesh_ip);
+    let ip_list = filtered_list.load::<String>(&conn)?;
+
+    for ip in ip_list {
+        let filtered_list = clients.select(internet_ipv6).filter(mesh_ip.eq(&ip));
+        let mut sub = filtered_list.load::<String>(&conn)?;
+
+        if sub.len() > 1 {
+            error!("More than one ipv6 for a given client");
```

Comment:
isnt' this an expected case when a client migrates? But I suppose we won't hit that on an initialization.

Reply:
My understanding was that each client would have only one subnet assigned for each exit cluster, but a different subnet for different exit clusters?

Reply:
If two exits within the same cluster have different ipv6 subnets a clietn may need to be assigned one of both to roam between them.